### PR TITLE
Useapi

### PR DESCRIPTION
**Pull request by davecramer**
Use api for kubernetes to fail over

### CODE REVIEW

File: plugins/kube/apiclient.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+	if err != nil {
+		return  fmt.Errorf("could not get pod info: %v", err)
+	}
+
+	if len(pod.Spec.Containers) != 1 {
+		return  fmt.Errorf("could not determine which container to use")
+	}
+
+	log.Info("using exec")
```
Comment (abrightwell) [inline plugins/kube/apiclient.go]: Can we be more specific with this log message? Is it really an Info message or should it be a Debug?

**Comment by davecramer** [comment]
Yes, I had no intention of not squashing the commits.

**Comment by davecramer** [comment]
I have solved this in a different way anyway. It will go away shortly

**Comment by abrightwell** [comment]
Sounds good. Though, probably worth keeping in the back of our heads, just in case. Seems like an `options`-like parameter might be something useful at some point. But, of course, only when it becomes a necessity.

**Comment by abrightwell** [comment]
Why was this merged without a completed review?

**Pull request merged**